### PR TITLE
OCPBUGS-33352: Fix PatternFly modules check

### PR DESCRIPTION
**Pull request by vojtechszocs**
Refactor code in `scripts/check-patternfly-modules.sh` to avoid relying on `yarn.lock` v1 format and parse `yarn why` command output instead.

- `check-resolution` function now checks for a single major version resolution of the given package
- this allows us to check that we use a single e.g. `@patternfly/react-core` version for both 4.x and 5.x
- `PKGS_TO_CHECK` array updated to include `@patternfly/patternfly` packages

x

How to reproduce the original issue (related to #13680) - open `frontend/package.json` and make the following change:
```diff
-    "@types/react-redux": "6.0.2",
+    "@types/react-redux": "7.1.33",
```
then run `yarn install` in `frontend` directory; the PatternFly modules check script will fail because the above change will trigger some other changes in `yarn.lock` file.

### CODE REVIEW

File: frontend/scripts/check-patternfly-modules.sh [hunk 1]
```diff
@@ -8,50 +8,68 @@ COL_YELLOW='\033[33m'
...
-  '@patternfly/react-log-viewer'
-  '@patternfly/react-styles'
-  '@patternfly/react-table'
-  '@patternfly/react-tokens'
-  '@patternfly/react-topology'
-  '@patternfly/react-user-feedback'
-  '@patternfly/react-virtualized-extension'
+  '@patternfly/patternfly:5'
+  '@patternfly/patternfly:4'
+  '@patternfly/quickstarts:5'
```
Comment (jhadvig) [inline frontend/scripts/check-patternfly-modules.sh:48]: for `quickstarts` we have [both version 4 and 5](https://github.com/openshift/console/blob/master/frontend/package.json#L142)... even though the version 4 is pointing to `patternfly/quickstarts@2.4.2`

**Review by jhadvig — COMMENTED** [review]
Thanks @vojtechszocs for the fix. Added one comment. PTAL

File: frontend/scripts/check-patternfly-modules.sh (hunk 1 above)
Comment (vojtechszocs) [reply frontend/scripts/check-patternfly-modules.sh:48]: @jhadvig `yarn why @patternfly/quickstarts` shows just one resolution
```
=> Found "@patternfly/quickstarts@5.1.0"
```
Our `@patternfly-4/*` aliased packages are excluded from this list - it only contains unaliased PatternFly packages that may appear in multiple versions due to some of them depending on other versions.

A good example of this is `@patternfly/patternfly` package - see `yarn why @patternfly/patternfly` resolutions
```
=> Found "@patternfly/patternfly@5.2.1"
=> Found "@patternfly-4/react-console#@patternfly/patternfly@4.139.2"
```
Notice the `@patternfly/patternfly@4.139.2` resolution that is pulled in transitively via `@patternfly-4/react-console` aliased package.

Comment (jhadvig) [reply frontend/scripts/check-patternfly-modules.sh:48]: Thanks for explanation @vojtechszocs 👍

**Pull request merged**